### ISSUE TITLE
setting minWidth dynamically not working

### ISSUE THREAD

**Issue by Mshardul** [ag-grid/ag-grid#4134]
**I'm submitting a ...**
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**

Not able to set the minWidth dynamically using setColumnDefs().

**Expected behavior**
setting minWidth through setColumnDefs() should work.

**Please tell us about your environment:**

tried this on both Windows and MacOS.

* **ag-Grid version:** 23.2.1

* **Browser:** Safari, Google Chrome

* **Language:** Angular

> this.columnDefs.forEach(colDef => { colDef.minWidth = 100; });
>     this.gridApi.setColumnDefs(this.columnDefs);

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by kiril-matev** [comment]
Hello!

Thank you for submitting this.

We have made a number of column attributes stateful, as per our documentation:
https://www.ag-grid.com/javascript-grid-column-updating-definitions/#changing-column-state

However, that's not the case yet for minWidth and maxWidth. We agree this would be a useful addition and have added this as a feature request.

We have added this feature request to our backlog and we are tracking it with the following reference and description:
**AG-4714 Make maxWidth & minWidth stateful attributes for columns**

You can check whether this item is selected for implementation by taking a look at the NEXT RELEASE tab of the product pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

I would recommend signing up for automatic release notifications for ag-Grid. Then, following a release, check whether the item was implemented by looking for it in the changelog page.

Thanks again for bringing this up with us.

Regards,

**Comment by makinggoodsoftware** [comment]
Hi,

In order to help us keep our issues queue clear, we are going to close this issue

The official way to check the status for an issue raised here is to check its AG-xxxx ID against our pipeline and changelog

…the pipeline (if not yet developed)

https://www.ag-grid.com/ag-grid-pipeline/

... or the changelog, (if the issue has been resolved)

https://www.ag-grid.com/ag-grid-changelog/

We keep issues raised in the pipeline through github open for a while so that the community can add some feedback to them, and close them when there is no activity for a while.

If you want to add further feedback to a closed ticket with this comment, we would recommend opening a new ticket as we don’t monitor closed issues.

**Comment by kiril-matev** [comment]
Hi @Mshardul,

I wanted to follow up on this.

We have since updated this behavior and you can now dynamically set column minWidth values.

In order to modify the column minWidth, you need to update the column definitions as shown below:
https://plnkr.co/edit/3Srk6YklVH4lJsGb
 
Click the button above the grid and note the column minWidth values change and are updated in the grid layout.

Thank you for your time and for using AG Grid!

**Comment by hinok** [comment]
AG-4714 is still placed in 

> …the pipeline (if not yet developed)
> https://www.ag-grid.com/ag-grid-pipeline/

According to my testing, it was fixed and released in `25.3.0`. I found that the dynamic setting on `minWidth` starts working on that specific version. Maybe it will help someone.
